modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -185,6 +185,21 @@ public ObjectField addOrUpdateCustomObjectField(
 				objectFieldSettings);
 		}
 
+		if (Objects.equals(businessType, "Relationship")) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinitionId);
```

Comment:
@larissacribeiro we have this same logic [here](https://github.com/liferay-objects/liferay-portal/blob/0e2895f791b4654956813ecd9e25f6621aede39e/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java#L170-L173).

Please move this logic to be outside the `if` statement to reuse code.